server/app/api.py

```diff
@@ -1987,6 +2010,72 @@ def new_entity(self, attributes):
         ent.assigned_staff = [models.User.get_or_insert(
             user.id()).key for user in ent.assigned_staff]
         return ent
+    
+    
+class QueuesAPI(APIResource):
+    """ API resource for sets of queues """
+    
+    contains_entities = False
+
+    methods = {
+        'generate': {
+            'methods': set(['POST']),
+            'web_args': {
+                'course': KeyArg('Course', required=True),
+                'assignment': KeyArg('Assignment', required=True)
+            }
+        }
+    }
+    
+    def generate(self, user, data):
+        """ Splits up submissions among staff members """
+        
+        self.check_permissions(user, data)
+        course, assignment = data['course'].get(), data['assignment'].get()
+        userify = lambda parts: [part.user.get() for part in parts]
+        
+        staff = userify(models.Participant.query(
```

Comment:
We don't necessarily want all staff grading composition. We'd need a selection of staff. The front-end could display a list of staff and the user could select the users to choose. 

Perhaps the create queues button goes a page `/assignId/queues/new` choosing what staff members get assigned. That page can then include the userIDs selected in it's post.